File: packages/rust-sdk/src/composition/idempotency.rs
use std::collections::HashMap;

pub fn with_idempotency(headers: &mut HashMap<String, String>, key: &str) {
    headers.insert("Idempotency-Key".to_string(), key.to_string());
}

pub fn with_idempotency_header(headers: &mut HashMap<String, String>, key: &str, header_name: &str) {
    headers.insert(header_name.to_string(), key.to_string());
}

File: packages/ruby-sdk/lib/strait/composition/idempotency.rb
# frozen_string_literal: true

module Strait
  module Composition
    def self.with_idempotency(headers, key)
      headers = (headers || {}).dup
      headers["Idempotency-Key"] = key
      headers
    end

    def self.with_idempotency_header(headers, key, header_name)
      headers = (headers || {}).dup
      headers[header_name] = key
      headers
    end
  end
end
